examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,135 @@
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void ReadBufferMap(
+	WGPUBufferMapAsyncStatus status, const uint8_t *data, uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        printf("Times: %s \n", data);
+    }
+}
+
+int main(int argc, char *argv[]) {
+
+    if (argc != 5) {
+        return 0;
+    }
+
+    int numbers[] = {
+        atoi(argv[1]),
+        atoi(argv[2]),
+        atoi(argv[3]),
+        atoi(argv[4]),
+    };
+
+    int size = sizeof(numbers);
+
+	uint32_t numbers_length = sizeof(numbers) / sizeof(int);
+
+    WGPUInstanceId instance = wgpu_create_instance();
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions =
+                {
+                    .anisotropic_filtering = false,
+                },
+        });
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
+        &(WGPUBufferDescriptor){.size = numbers_length,
+            .usage = WGPUBufferUsage_MAP_READ | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC},
+        numbers);
+
+    WGPUBufferId storage_buffer = wgpu_device_create_buffer(device,
+        &(WGPUBufferDescriptor){.size = size,
+            .usage = WGPUBufferUsage_STORAGE | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC});
+
+    WGPUBindGroupLayoutId bind_group_layout =
+        wgpu_device_create_bind_group_layout(device,
+            &(WGPUBindGroupLayoutDescriptor){
+                .bindings = &(WGPUBindGroupLayoutBinding){.binding = 0,
+                    .visibility = WGPUShaderStage_COMPUTE,
+                    .ty = WGPUBindingType_StorageBuffer}});
+
+    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
+        &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
+            .bindings = &(WGPUBindGroupBinding){
+				.binding = 0
+	}});
+
+
+    WGPUPipelineLayoutId pipeline_layout =
+        wgpu_device_create_pipeline_layout(device,
+            &(WGPUPipelineLayoutDescriptor){
+                .bind_group_layouts = &bind_group_layout});
+
+    WGPUShaderModuleId shader_module = wgpu_device_create_shader_module(device,
+        &(WGPUShaderModuleDescriptor){
+            .code = read_file("./../../data/shader.comp.spv")});
+
+	WGPUComputePipelineId compute_pipeline =
+			wgpu_device_create_compute_pipeline(device,
+				&(WGPUComputePipelineDescriptor){
+					.layout = pipeline_layout,
+					.compute_stage = &(WGPUPipelineStageDescriptor){
```

Comment:
according to @grovesNL , this is the problem. I'm puzzled as to why it even compiles if it's wrong?..

Reply:
Yeah, this is the first issue I noticed. It does compile but it has a warning:

`warning: incompatible pointer to integer conversion initializing
      'WGPUIndex' (aka 'unsigned int') with an expression of type 'WGPUPipelineStageDescriptor *' [-Wint-conversion]`

Reply:
Thank you!
wait, why is it saying that "WGPUIndex" is being initialized? It should be `WGPUPipelineStageDescriptor` there.

Reply:
The error message is a bit misleading, probably because of the nested designated initializers. I think it's talking about the first field it hits which happens to be `WGPUIndex` (the first field of `WGPUShaderModuleId`/`WGPUId`).

The fix should (still) be to remove `&` here though